Framework/Algorithms/src/MaskBins.cpp

```diff
@@ -75,40 +61,28 @@ void MaskBins::exec() {
     throw std::invalid_argument(msg.str());
   }
 
-  //---------------------------------------------------------------------------------
-  // what spectra (workspace indices) to load. Optional.
-  this->spectra_list = this->getProperty("SpectraList");
-  if (!this->spectra_list.empty()) {
-    const int numHist = static_cast<int>(inputWS->getNumberHistograms());
-    //--- Validate spectra list ---
-    for (auto wi : this->spectra_list) {
-      if ((wi < 0) || (wi >= numHist)) {
-        std::ostringstream oss;
-        oss << "One of the workspace indices specified, " << wi
-            << " is above the number of spectra in the workspace (" << numHist
-            << ").";
-        throw std::invalid_argument(oss.str());
-      }
-    }
+  // Copy indices from legacy property
+  std::vector<int> spectraList = this->getProperty("SpectraList");
+  if (!spectraList.empty()) {
+    setProperty("InputWorkspaceIndexSet", spectraList);
```

Comment:
You are not preventing users from supplying information to both properties? I think you should probably do so.

Reply:
You might also consider using [EnableWhenProperty](https://github.com/mantidproject/mantid/blob/master/Framework/Kernel/src/EnabledWhenProperty.cpp) for completeness in GUI for this algorithm.